components/vision.py

```diff
@@ -1,213 +1,161 @@
-        self.field_pos_obj_right = self.field.getObject("vision_pose_right_cam")
-        self.pose_log_entry = wpiutil.log.FloatArrayLogEntry(
-            self.data_log, "vision_pose"
-        )
+    def on_enable(self) -> None:
+        self.add_to_estimator = True
 
     def execute(self) -> None:
-        for cam_idx, (camera, trans) in enumerate(self.cameras):
-            # if results didnt see any targets
-            if not (results := camera.getLatestResult()).hasTargets():
-                continue
-
-            # if we have already processed these results
-            timestamp = results.getTimestamp()
-            if timestamp == self.last_timestamps[cam_idx] and wpilib.RobotBase.isReal():
+        # if results didn't see any targets
+        if not (results := self.camera.getLatestResult()).hasTargets():
+            return
+
+        # if we have already processed these results
+        timestamp = results.getTimestamp()
+        if timestamp == self.last_timestamp and wpilib.RobotBase.isReal():
+            return
+        self.last_timestamp = timestamp
+
+        # old results cause pose estimator to crash and aren't very useful anyway
+        if abs(wpilib.Timer.getFPGATimestamp() - timestamp) > 0.5:
+            return
+
+        for target in results.getTargets():
+            poses = estimate_poses_from_apriltag(self.camera_to_robot, target)
+            if poses is None:
+                # tag doesn't exist
                 continue
-            self.last_timestamps[cam_idx] = timestamp
+            best_pose, alt_pose, self.last_pose_z = poses
+            pose = choose_pose(
+                best_pose,
+                alt_pose,
+                self.chassis.get_pose(),
+                target.getPoseAmbiguity(),
+            )
 
-            # old results cause pose estimator to crash and arent very useful anyway
-            if abs(wpilib.Timer.getFPGATimestamp() - timestamp) > 0.5:
+            # filter out likely bad targets
+            if target.getPoseAmbiguity() > 0.25 or target.getYaw() > 20:
                 continue
 
-            for target in results.getTargets():
-                _poses = estimate_poses_from_apriltag(trans, target)
-                # tag dosent exist
-                if _poses is None:
+            change = self.chassis.get_pose().translation().distance(pose.translation())
+            if change > 1.0:
+                self.rejected_in_row += 1
+                if self.rejected_in_row < 20:
                     continue
-                *poses, z1 = _poses
-                if cam_idx:
-                    self.last_z_1 = z1
-                else:
-                    self.last_z_2 = z1
-                pose = choose_pose(
-                    poses[0],
-                    poses[1],
-                    self.chassis.get_pose(),
-                    target.getPoseAmbiguity(),
+            else:
+                self.rejected_in_row //= 2
```

Comment:
Previously `rejected_in_row` would theoretically just go through exponential decay, never reaching 0, which was probably not intended.